Makefile.docker

```diff
@@ -8,7 +8,8 @@ docker-cilium-image-for-developers:
 	# a dedicated dockerignore file per Dockerfile.
 	$(QUIET)DOCKER_BUILDKIT=1 $(CONTAINER_ENGINE) build \
 		$(DOCKER_FLAGS) \
-		--build-arg LOCKDEBUG=\
+		--build-arg LOCKDEBUG=${RACE}\
+		--build-arg RACE=${RACE}\
```

Comment:
Might be helpful to put a note as to why `LOCKDEBUG=RACE` here because that is not the case for other commands.